akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ReplayingEvents.scala

```diff
@@ -131,93 +131,94 @@ private[akka] final class ReplayingEvents[C, E, S](
+                  Some((meta, replication.replicaId, replication))
+                case None => None
               }
 
-              replicatedMetaAndSelfReplica match {
-                case Some((meta, selfReplica, replication)) if meta.originReplica != selfReplica =>
-                  // keep track of highest origin seqnr per other replica
-                  state = state.copy(
-                    state = newState,
-                    eventSeenInInterval = true,
-                    version = meta.version,
-                    seenSeqNrPerReplica = state.seenSeqNrPerReplica + (meta.originReplica -> meta.originSequenceNr))
-                  replication.clearContext()
-                case Some((_, _, replication)) =>
-                  replication.clearContext()
-                  state = state.copy(state = newState, eventSeenInInterval = true)
-                case _ =>
-                  state = state.copy(state = newState, eventSeenInInterval = true)
-              }
+            val newState = setup.eventHandler(state.state, event)
+
+            setup.replication match {
+              case Some(replication) =>
+                replication.clearContext()
+              case None =>
             }
 
-            eventSeq match {
-              case SingleEventSeq(event) => handleEvent(event)
-              case EventsSeq(events)     => events.foreach(handleEvent)
-              case EmptyEventSeq         => // no events
+            replicatedMetaAndSelfReplica match {
+              case Some((meta, selfReplica, replication)) if meta.originReplica != selfReplica =>
+                // keep track of highest origin seqnr per other replica
+                state = state.copy(
+                  state = newState,
+                  eventSeenInInterval = true,
+                  version = meta.version,
+                  seenSeqNrPerReplica = state.seenSeqNrPerReplica + (meta.originReplica -> meta.originSequenceNr))
+                replication.clearContext()
+              case Some((_, _, replication)) =>
+                replication.clearContext()
+                state = state.copy(state = newState, eventSeenInInterval = true)
+              case _ =>
+                state = state.copy(state = newState, eventSeenInInterval = true)
             }
+          }
 
-            this
-          } catch {
-            case NonFatal(ex) =>
-              state = state.copy(repr.sequenceNr)
-              onRecoveryFailure(ex, eventForErrorReporting.toOption)
+          eventSeq match {
+            case SingleEventSeq(event) => handleEvent(event)
+            case EventsSeq(events)     => events.foreach(handleEvent)
+            case EmptyEventSeq         => // no events
           }
 
-        case RecoverySuccess(highestJournalSeqNr) =>
+          this
+        } catch {
+          case NonFatal(ex) =>
+            state = state.copy(repr.sequenceNr)
+            onRecoveryFailure(ex, eventForErrorReporting.toOption, "replaying-event")
```

Comment:
interesting, onRecoveryFailure will always end with:
```
throw new JournalFailureException(msg, cause)
```

and then we caught that again in the outer catch, and called onRecoveryFailure.

Doesn't make sense with that double handling, so this should be fine.